src/PockyBot.NET/Services/Triggers/Unpeg.cs

```diff
@@ -0,0 +1,309 @@
+using GlobalX.ChatBots.Core.People;
+using Microsoft.Extensions.Logging;
+using PockyBot.NET.Constants;
+using PockyBot.NET.Persistence.Models;
+using PockyBot.NET.Services.Helpers;
+
+namespace PockyBot.NET.Services.Triggers
+{
+    internal class Unpeg : ITrigger
+    {
+        private readonly IRandomnessHandler _randomnessHandler;
+        private readonly IChatHelper _chatHelper;
+        private readonly IBackgroundTaskQueue _backgroundTaskQueue;
+        private readonly IAsyncDelayer _asyncDelayer;
+        private readonly ILogger<Unpeg> _logger;
+        private readonly string[] _separatorWords = { "for", "because" };
+
+        public string Command => Commands.Unpeg;
+
+        public bool DirectMessageAllowed => false;
+
+        public bool CanHaveArgs => true;
+
+        public Role[] Permissions => Array.Empty<Role>();
+
+        public Unpeg(IRandomnessHandler randomnessHandler, IChatHelper chatHelper, IBackgroundTaskQueue backgroundTaskQueue, IAsyncDelayer asyncDelayer, ILogger<Unpeg> logger)
+        {
+            _randomnessHandler = randomnessHandler;
+            _chatHelper = chatHelper;
+            _backgroundTaskQueue = backgroundTaskQueue;
+            _asyncDelayer = asyncDelayer;
+            _logger = logger;
+        }
+
+        public Task<Message> Respond(Message message)
+        {
+            var recipientIsPerson = true;
+            var recipientName = message.MessageParts[1].Text.Trim().Substring(Command.Length).Trim();
+            string recipientId = null;
+
+            if (string.IsNullOrEmpty(recipientName))
+            {
+                if (message.MessageParts.Length < 3 || message.MessageParts[2].MessageType != MessageType.PersonMention)
+                {
+                    return Task.FromResult(new Message
+                    {
+                        Text = "Please provide someone or something to unpeg."
+                    });
+                }
+
+                recipientName = message.MessageParts[2].Text;
+                recipientId = message.MessageParts[2].UserId;
+            }
+            else
+            {
+                recipientIsPerson = false;
+                recipientName = GetRecipientNameFromString(recipientName);
+                if (recipientName == "me")
+                {
+                    recipientIsPerson = true;
+                    recipientName = message.Sender.Username;
+                    recipientId = message.Sender.UserId;
+                }
+            }
+
+            var response = SendRandomResponse(message.Sender, recipientIsPerson, recipientName, recipientId, message.RoomId);
+            return Task.FromResult(new Message
+            {
+                MessageParts = response
+            });
+        }
+
+        private MessagePart[] SendRandomResponse(Person sender, bool recipientIsPerson, string recipientName,
+            string recipientId, string roomId)
+        {
+            var random = _randomnessHandler.Random.Next(7);
```

Comment:
I wonder if something like [the solution proposed in this SO answer](https://stackoverflow.com/a/34279195) might be neater?

Reply:
Good idea. done